### PR TITLE
MRG: In Report.add_raw(), re-use the existing figure when creating butterfly plots of slices

### PR DESCRIPTION
**Pull request by hoechenberger**
Performance changes **EDIT I've updated the numbers after adding dde9732**

```
main: 12.3 s ± 294 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
pr:   6.98 s ± 429 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Because we're re-using the same figure, I could also get rid of the explicit `scalings` pre-calculation.

MWE:
```python
# %%
from pathlib import Path
import mne


sample_dir = Path(mne.datasets.sample.data_path())
sample_fname = sample_dir / 'MEG' / 'sample' / 'sample_audvis_raw.fif'

raw = mne.io.read_raw_fif(sample_fname, preload=True)
raw.crop(tmax=60)
r = mne.Report()

# %%
%timeit r.add_raw(raw=raw, title='foo')
```

x-ref #10107

### CODE REVIEW

File: tutorials/intro/70_report.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 statistics, MRI slices with overlaid BEM shells, all the way up to plots of
 estimated cortical activity.
 
-Compared to a Jupyter notebook, `mne.Report` is easier to deploy (the HTML
+Compared to a Jupyter notebook, `mne.Report` is easier to deploy (the HTMLf
```
Comment (agramfort) [inline tutorials/intro/70_report.py:15]:
```suggestion
Compared to a Jupyter notebook, `mne.Report` is easier to deploy (the HTML
```

File: mne/report/report.py [hunk 2]
```diff
@@ -2567,14 +2564,18 @@ def _render_raw_butterfly_segments(
...
-            for t in times:
-                tmin = max(t - 0.5, 0)
-                tmax = min(t + 0.5, raw.times[-1])
-                duration = tmax - tmin
-                fig = raw.plot(
-                    butterfly=True, show_scrollbars=False, start=tmin,
-                    duration=duration, scalings=scalings, show=False
-                )
+
+            # Create the figure once and re-use it for performance resons
```
Comment (agramfort) [inline mne/report/report.py:2568]:
```suggestion
            # Create the figure once and re-use it for performance reasons
```

**Comment by hoechenberger** [comment]
`_render_slider()` now has a new `imgs` kwarg in addition to the old `figs`, which allows to pass images directly instead of only MPL figures. This allows us to take "snapshots" of the same figure in the `for` loop without having to create a copy first.

**Comment by hoechenberger** [comment]
Rendered tutorial:
https://39608-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/intro/70_report.html#adding-raw-data

**Review by agramfort — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks @hoechenberger !

**Pull request merged**